addons/website_event/controllers/main.py

```diff
@@ -71,6 +77,14 @@ def sd(date):
 
         if searches['search']:
             domain_search['search'] = [('name', 'ilike', searches['search'])]
+        if search_tags:
+            grouped_tags = defaultdict(list)
+            for tag in search_tags:
+                grouped_tags[tag.category_id].append(tag)
+            group_domain_list = []
+            domain_search['tags'] = []
+            for group in grouped_tags:
+                domain_search['tags'] = expression.AND([domain_search['tags'], [('tag_ids', 'in', [tag.id for tag in grouped_tags[group]])]])
```

Comment:
Example: You filter on age: 10-12 and activity: football.

By using your solution, you would get all events that are for tagged "age: 10-12" and all events that are tagged "activity: football".

Doing it my way allows to only get events who are tagged "age: 10-12" AND "activity: football".
Add another tag "age: 12-15" to the search and it would fetch the ones who are tagged:
("age: 10-12" OR "age: 12-15")  AND "activity: football" 
IMO this is what should be done when filtering on tags.

Let me know if my explanations are clear enough